scala-package/infer/src/test/scala/ml/dmlc/mxnet/infer/ImageClassifierSuite.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ml.dmlc.mxnet.infer
+
+import ml.dmlc.mxnet._
+import ml.dmlc.mxnet.{DataDesc, Shape}
+
+import org.mockito.Matchers._
+import org.mockito.Mockito
+import org.scalatest.{BeforeAndAfterAll}
+
+// scalastyle:off
+import java.awt.image.BufferedImage
+// scalastyle:on
+
+/**
+  * Unit tests for ImageClassifier
+  */
+class ImageClassifierSuite extends ClassifierSuite with BeforeAndAfterAll {
+
+  class MyImageClassifier(modelPathPrefix: String,
+                           inputDescriptors: IndexedSeq[DataDesc])
+    extends ImageClassifier(modelPathPrefix, inputDescriptors) {
+
+    override def getPredictor(): MyClassyPredictor = {
+      Mockito.mock(classOf[MyClassyPredictor])
+    }
+
+    override def getClassifier(modelPathPrefix: String, inputDescriptors:
+    IndexedSeq[DataDesc]): Classifier = {
+      Mockito.mock(classOf[Classifier])
+    }
+
+    def getSynset(): IndexedSeq[String] = synset
+  }
+
+  test("ImageClassifierSuite-testRescaleImage") {
+    val image1 = new BufferedImage(100, 200, BufferedImage.TYPE_BYTE_GRAY)
+    val image2 = ImageClassifier.reshapeImage(image1, 1000, 2000)
+
+    assert(image2.getWidth === 1000)
```

Comment:
Could you execute this test once, generate an MD5 hash based on that output and verify it here? The shape will probably always be right, but the question is whether the actual content has been resized correctly or the code just generated some random data.

Reply:
Python has libraries to do image hashing(imagehash). Getting hash is not that trivial in Java. Will have to implement image hashing function (average_hash)to match the hash.

However, I ran the function "ImageClassifier.reshapeImage" and tested the hash with python if it's resizing correctly or not. It's working as expected.

Reply:
@marcoabreu I am not convinced that it is worth the effort and extra code to test this functionality. see this https://www.pyimagesearch.com/2017/11/27/image-hashing-opencv-python/. 
Also, the reshaped images are used to test against the model for inference and we verify the output(label) matches what the expected output.

Reply:
Oh I see, validating them in a follow up step with the output labels is even better